### PR TITLE
[Issue-180] Make Healthy L1 and L2 RPC Endpoints available to jobs in CircleCI

### PR DESCRIPTION
**Pull request by twoshark**
**Description**

- This PR adds CircleCI Jobs to test rpc endpoints for l1 & l2 for sepolia, goerli and mainnet
- The jobs define RPC urls for all networks to be loaded by subsequent jobs
- An example mainnet job is included that sets the required L1 & L2 RPC env vars
- The endpoints are verified by confiriming the rpc endpoint can be reached, that the replica is not syncing and that its block height is greater than 0. 

**Tests**

See [Checks](https://github.com/ethereum-optimism/superchain-ops/pull/22/checks)

**Metadata**

- [Client Pod Issue 180](https://github.com/ethereum-optimism/client-pod/issues/180)

### CODE REVIEW

File: ops/verify-geth-endpoint.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+data='{"jsonrpc":"2.0","method":"eth_syncing","params":[],"id":1}'
+response=$(curl -s -X POST -H "Content-Type: application/json" --data "$data" $URL)
+echo $response
+syncing=$(echo $response | jq -r '.result')
+if [ "$syncing" != "false" ]; then
+  echo "$URL is syncing."
+  exit 1
+fi
+
+echo "RPC endpoint at $URL is healthy."
```
Comment (twoshark) [inline ops/verify-geth-endpoint.sh:19]:
```suggestion
echo "RPC endpoint at $URL is healthy."

```

File: .github/workflows/define_rpc_endpoints.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    if: ${{ needs.check_rpc_endpoints.outputs.l1_rpc_url && needs.check_rpc_endpoints.outputs.l2_rpc_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Use RPC URLs
+        run: |
+          L1_RPC_URL=${{ needs.check_rpc_endpoints.outputs.l1_rpc_url }}
+          L2_RPC_URL=${{ needs.check_rpc_endpoints.outputs.l2_rpc_url }}
+          echo "L1 RPC URL: $L1_RPC_URL"
+          echo "L2 RPC URL: $L2_RPC_URL"
+          # Use L1_RPC_URL and L2_RPC_URL here.
```
Comment (twoshark) [inline .github/workflows/define_rpc_endpoints.yaml:56]:
```suggestion
          # Use L1_RPC_URL and L2_RPC_URL here.

```

File: .gitignore [hunk 3]
```diff
@@ -16,3 +16,4 @@ lib/
 cache/
 
 .DS_Store
+.idea/
```
Comment (twoshark) [inline .gitignore:19]:
```suggestion
.idea/

```

File: .circleci/config.yaml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+          
+
+workflows:
+  version: 2
+  build_and_test_workflow:
+    jobs:
+      - check_rpc_endpoints
+      - example_mainnet_job:
+          requires:
+            - check_mainnet_rpc_endpoints
```
Comment (twoshark) [inline .circleci/config.yaml:131]:
```suggestion
            - check_mainnet_rpc_endpoints

```

File: .circleci/config.yaml [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    default: 5
+
+jobs:
+  check_sepolia_rpc_endpoints:
+    docker:
+      - image: us-docker.pkg.dev/oplabs-tools-artifacts/images/ci-builder:latest
+    steps:
+      - checkout
+      - run:
+          name: Install JQ
```
Comment (mslipper) [inline .circleci/config.yaml:33]: I think JQ is already on CI-builder.

**Review by mslipper — APPROVED** [review]

File: ops/verify-geth-endpoint.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+set -euo pipefail
+# Script to check the syncing status and block timestamp of an Ethereum RPC endpoint
```
Comment (twoshark) [inline ops/verify-geth-endpoint.sh:4]:
```suggestion
```

**Review by mslipper — APPROVED** [review]

**Pull request merged**